src/bin/pgcopydb/copydb_schema.c

```diff
@@ -542,7 +542,8 @@ copydb_fetch_source_schema(CopyDataSpec *specs, PGSQL *src)
 		!sourceDB->sections[DATA_SECTION_TABLE_DATA].fetched)
 	{
 		/* copydb_fetch_filtered_oids() needs the table size table around */
-		if (!schema_prepare_pgcopydb_table_size(src,
+		if (!specs->estimateTableSizes &&
+			!schema_prepare_pgcopydb_table_size(src,
```

Comment:
I think it's time to move that logic within the `copydb_prepare_table_specs()` function. What do you think?